ProjFS.Linux/PrjFSLib.Linux.Managed/Interop/ProjFS.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Runtime.InteropServices;
+using Mono.Unix.Native;
+
+namespace PrjFSLib.Linux.Interop
+{
+    internal class ProjFS
+    {
+        // TODO(Linux): set value from that defined in Linux library header
+        public const int PlaceholderIdLength = 128;
+
+        public const ulong PROJFS_CLOSE_WRITE = 0x00000008;
```

Comment:
With these values and the one above, I wonder if we could do something like is done in [`GenerateGitVersionConstants.sh`](https://github.com/Microsoft/VFSForGit/blob/master/Scripts/Mac/GenerateGitVersionConstants.sh) -- i.e., either extract these values from the C header file using `sed`, `grep`, etc., or build a small C program which includes the header and when run spits out the values, which we then put into a small C# `ProjFS.Linux.Constants.cs` file or something.  Does that seem out of bounds?

Reply:
No, that seems very apt. We might have to do it with a C program because of definitions like:

```c
#define himask(x)    (((uint64_t)x) << 32)
…
#define PROJFS_CREATE_SELF   himask(0x0001)  /* File was created */
```

I'll have a think about this, but yes, you're quite right -- this is fragile and we can do better.

Reply:
Once I work this out, I'll open a PR against the Microsoft upstream repo and invite review from our friends there, as well as yours again on the newest changes.